llvm/lib/CodeGen/GlobalMerge.cpp

```diff
@@ -713,6 +714,25 @@ bool GlobalMergeImpl::run(Module &M) {
     // Ignore all "required" globals:
     if (isMustKeepGlobalVariable(&GV))
       continue;
+    auto checkUsers = [](const GlobalVariable *GV) {
+      for (const User *CurrentUser : GV->users()) {
+        auto *I = dyn_cast<Instruction>(CurrentUser);
+        if (!I)
+          continue;
+        // Do not merge globals in exception pads.
+        if (I->isEHPad())
```

Comment:
Isn't this already handled in https://github.com/llvm/llvm-project/blob/408abb2353fdc359de09d2ba3d3521ce2b5fd0b1/llvm/lib/CodeGen/GlobalMerge.cpp#L636? All code for globals that cannot be merged based on uses should be kept in there.